HowToMLFlow.md

```diff
@@ -0,0 +1,234 @@
+
+The **MLproject file**, this file is kind of like the description of the pipeline and it helps MLFlow understand how your pipeline is run and how to show the runs in the UI
+
+### Creating the steps
+
+To create the steps you of course need to create the python files for the pipeline to use. In this case let's create 3 steps, called each: main.py, data_ingestion.py, data_display.py.
+
+Now that you have created these 3 steps, it's time to make the main.py useful by making it the orchestrator/controller of the 2 other steps execution.
+
+Using the mlflow.run() will make a specific run with the help of MLFlow and the MLproject, and then the files themselves start the run they are assigned using mlflow.start_run()
+
+```python
+data_ingestion_run = mlflow.run(
+            "",
+            "data_ingestion.py",
+            parameters={},
+            env_manager="local",
+        )
+
+data_display_run = mlflow.run(
+            "",
+            "data_display.py",
+            parameters={"data_ingest_run_id": data_ingestion_run.run_id},
+            env_manager="local",
+        )
+```
+
+The parameter is explained in the [MLproject section](#mlproject-file)
+
+### Content for data_ingestion step
+
+Now onto what we want to include in the steps.
+
+For the data ingestion, we want to get the dataset from a url and save it as an artifact.
+To do that we will need to use these libraries:
+
+- Pandas
+- MLFlow
+
+As soon as you have imported the above mentioned libraries, we can then write the code for downloading the dataset.
+
+The url is something you can change to whatever dataset you might want.
+
+```python
+def task():
+    with mlflow.start_run() as mlrun:
+        url= https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data
+
+        df = pd.read_csv(url)
+        df.to_csv("path_to_save_to")
+```
+
+The code above will download and read the dataset from the url and make it into a pandas DataFrame. The dataframe can then be saved to a specific place of your choosing.
+
+Now for saving/logging it to the mlflow run itself, we want to save it in a place we know the path to.
+
+```python
+mlflow.log_artifact("path_to_csv_file")
+```
+
+Now the artifact will be saved in the data_ingestion mlflow run, to be able to use it in the next run you'll have to access the data_ingestion run_id.
+
+To do this you will need to know how to work the MLproject file, but first we also have to make sure you have a **conda environment** file setup so you can run the project when it's done.
+
+### Making a conda_env
+
+To make a **conda environment** file you will need to understand how it's layout looks.
+
+Here is an example. A more detailed tutorial for conda_env can be found [here](https://blog.devgenius.io/using-conda-environments-for-python-all-you-need-to-know-2eb36e224d1c)
+
+![conda environment example](Conda_env_example.PNG)
```

Comment:
Instead of an image, paste the actual code as you did below.
Also, it's redundant to have 2 exmaples, you can shorten this.